python/mlc_chat/serve/config.py

```diff
@@ -22,6 +22,8 @@ class GenerationConfig:
 
     max_new_tokens : int
         The maximum number of generated tokens.
+        It may be negative, which means the generation will not stop
```

Comment:
Is this design consistent with OpenAI's?

Reply:
OpenAI supports `max_tokens = None` https://platform.openai.com/docs/api-reference/chat/create#chat-create-max_tokens, which means
> The total length of input tokens and generated tokens is limited by the model's context length.
 

In our case, “being negative means `None` in OpenAI” if we don’t use `std::optional`. Being negative is clear from my POV. I am not sure if using `std::optional` will incur additional runtime overhead.

Reply:
I agree at the interface level we should just be consistent. aka support max_tokens = None and rename it to max_tokens

Reply:
Updated accordingly. Now we use the name `max_tokens` and supports None in Python. On C++ side when we load the config from JSON, we check if the field value is `null`, and if so we use `-1` for `max_tokens` in C++.